pkg/net/host.go

```diff
@@ -129,7 +128,12 @@ func ParseHost(s string) (*Host, error) {
 		isPortSet = true
 	}
 
+	// IPv6 requires a link-local address on every network interface.
+	// `%interface` should be preserved.
+	nHost := host
+
 	if i := strings.Index(host, "%"); i > -1 {
+		// `%interface` can be skipped for validity check though.
```

Comment:
You are not using `host` below, hence you wouldn't need `nHost`

Reply:
done. modified

Reply:
done. modified this as discussed